### PR TITLE
NAS-127060 / 24.10 / Read only user able to edit Adv > Access Settings > token lifetime

### PR DESCRIPTION
**Pull request by AlexKarpov98**
Testing: 
see ticket & use readonly user

### CODE REVIEW

**Review by RehanY147 — APPROVED** [review]

File: src/app/pages/system/advanced/access/access-form/access-form.component.ts [hunk 1]
```diff
@@ -40,6 +42,7 @@ export class AccessFormComponent implements OnInit {
   }
 
   protected readonly Role = Role;
+  readonly requiresRoles = [Role.FullAdmin];
```
[the remark this answers is not in the document]
Comment (AlexKarpov98) [reply src/app/pages/system/advanced/access/access-form/access-form.component.ts:45]: @RehanY147 - I agree, I just use same naming as we have, but I can update all of them easily.

Screenshot 2024-02-01 at 11 48 24

File: src/app/pages/system/advanced/access/access-form/access-form.component.ts [hunk 2]
```diff
@@ -69,28 +73,30 @@ export class AccessFormComponent implements OnInit {
   }
 
   onSubmit(): void {
-    this.store$.dispatch(lifetimeTokenUpdated({ lifetime: this.form.value.token_lifetime }));
+    this.authService.hasRole(this.requiresRoles).pipe(filter(Boolean), untilDestroyed(this)).subscribe(() => {
```
[the remark this answers is not in the document]
Comment (AlexKarpov98) [reply src/app/pages/system/advanced/access/access-form/access-form.component.ts:76]: > We can break this down into a separate method. So it looks something like the following `this.canProceed().pipe(switchMap(() => this.isEnterprise ? this.updateSystem() : of())).subscribe(...)`

don't forget that empty `of()` will not emit any data. 😄

**Review by RehanY147 — CHANGES_REQUESTED** [review]
I see different values on the form for `FullAdmin` user or `ReadOnly` user.
See 300 for `ReadOnly` user and 300000 for `FullAdmin` user.

**Comment by AlexKarpov98** [comment]
> I see different values on the form for `FullAdmin` user or `ReadOnly` user. See 300 for `ReadOnly` user and 300000 for `FullAdmin` user.
@RehanY147 
It's not possible for NON ADMIN user to submit the form at all!
How is that possible? (cache?)

**Comment by AlexKarpov98** [comment]
> I see different values on the form for `FullAdmin` user or `ReadOnly` user. See 300 for `ReadOnly` user and 300000 for `FullAdmin` user.

@RehanY147 - I see it as well, but I think it's out of the scope for this ticket and maybe it's a Middleware Bug?

Screenshot 2024-02-01 at 12 03 12
Screenshot 2024-02-01 at 12 03 21

**Comment by AlexKarpov98** [comment]
@RehanY147 - that's fine. Confirmed by @yocalebo!
**Caleb St. John  [2:15 PM]**
Hi, what is the issue? That’s not a global value, it’s tied to each credential I’m pretty sure

**Review by RehanY147 — APPROVED** [review]

**Pull request merged**